### PR TITLE
Created a User Profile view and its landing page

### PR DESCRIPTION
**Pull request by shrey1608**
User Profile  and its features:
- Created a landing page for userprofile
- Added profile view and profile update view 
- As the user will need to put his details in the profile so if the user is new or he hasnt created his profile then nothing would be displayed and add button would be displayed and after he has added it the update profile button would be displayed
- update profile and add profile forms
- update profile page

### CODE REVIEW

File: fpage/models.py [hunk 1]
```diff
@@ -21,4 +21,13 @@ class Song(models.Model):
...
-        return self.song_title 
\ No newline at end of file
+        return self.song_title 
+
+class UserProfile(models.Model):
+    user = models.ForeignKey(User,on_delete=models.CASCADE)
+    first_name=models.CharField(max_length=50)
+    last_name=models.CharField(max_length=50)
+    email=models.CharField(max_length=150)
+    phone_num=models.IntegerField()
```
Comment (ExpressHermes) [inline fpage/models.py]: Provide phone number validation

File: fpage/models.py [hunk 2]
```diff
@@ -21,4 +21,13 @@ class Song(models.Model):
     is_favorite = models.BooleanField(default=False)
 
     def __str__ (self):
-        return self.song_title 
\ No newline at end of file
+        return self.song_title 
+
+class UserProfile(models.Model):
+    user = models.ForeignKey(User,on_delete=models.CASCADE)
```
Comment (ExpressHermes) [inline fpage/models.py]: This should be a OneToOne field as one user can have only one profile

File: fpage/views.py [hunk 3]
```diff
@@ -66,13 +66,51 @@ def logoutUser(request):
...
+    profile= get_object_or_404(UserProfile, user=request.user)
+    form = ProfileForm(initial={'firstname': profile.first_name , 'lastname': profile.last_name, 'email': profile.email, 'phone_num': profile.phone_num, 'fav_genre':profile.fav_genre , 'profile_image':profile.profile_image})
+    if request.method == "POST":  
+        form = ProfileForm(request.POST, instance=profile)  
+        if form.is_valid():  
+            try: 
+                form.save() 
+                model = form.instance
+                return redirect('fpage:profile')
+            except Exception as e:
```
Comment (ExpressHermes) [inline fpage/views.py]: Instead of passing the exception  try to show error messages

File: fpage/views.py [hunk 4]
```diff
@@ -66,13 +66,51 @@ def logoutUser(request):
...
+def profileCreate(request):
+    if request.method == "POST":  
+            form = ProfileForm(request.POST)
+            if form.is_valid():  
+                try:  
+                    fs=form.save(commit=False)
+                    fs.user=request.user
+                    fs.save()
+                    return redirect('fpage:profile')  
+                except:
```
Comment (ExpressHermes) [inline fpage/views.py:103]: Handle error messages properly

**Review by ExpressHermes — COMMENTED** [review]
Resolve the issues

File: fpage/views.py (hunk 3 above)
Comment (shrey1608) [reply fpage/views.py]: @ExpressHermes Like this?
image

Comment (ExpressHermes) [reply fpage/views.py]: Also provide proper errors `form.errors` to send the message for why the updation/creation failed

Comment (ExpressHermes) [reply fpage/views.py]: Also try to send these in normal PR comments as viewing them here is inconvenient

Comment (shrey1608) [reply fpage/views.py]: yes i by mistake did it here

**Comment by shrey1608** [comment]
@ExpressHermes Please review it

**Comment by shrey1608** [comment]
@ExpressHermes Any updates?

**Comment by ExpressHermes** [comment]
I will ping you after reviewing it. Will try to review it tonight

**Comment by shrey1608** [comment]
Sure!

File: fpage/models.py [hunk 5]
```diff
@@ -24,10 +25,10 @@ def __str__ (self):
         return self.song_title 
 
 class UserProfile(models.Model):
-    user = models.ForeignKey(User,on_delete=models.CASCADE)
+    user = models.OneToOneField(User,on_delete=models.CASCADE)
     first_name=models.CharField(max_length=50)
     last_name=models.CharField(max_length=50)
     email=models.CharField(max_length=150)
-    phone_num=models.IntegerField()
+    phone_num=PhoneNumberField(null=False, blank=False, unique=True)
```
Comment (ExpressHermes) [inline fpage/models.py]: Make the phone number field both `null=True` and `blank=True`. Phone number should be a optional field.

Comment (shrey1608) [reply fpage/models.py]: ok

**Comment by shrey1608** [comment]
Please merge it i have another PR to do Thanks @ExpressHermes

**Review by ExpressHermes — APPROVED** [review]

**Comment by shrey1608** [comment]
Please provide points for this @ExpressHermes

**Comment by ExpressHermes** [comment]
give @shrey1608 150 points

**Pull request merged**